src/main/java/daos/DBPostDataAccessObject.java

```diff
@@ -0,0 +1,199 @@
+
+        try {
+            UpdateResult result = this.postRepository.updateOne(query, updates, insertNewDoc);
+            System.out.println("Upserted id: " + result.getUpsertedId());
+        } catch (MongoException error) {
+            // throw err?
+        }
+    }
+
+    
+    /**
+     * Inserts the given post into the database.
+     * @param user - a user in the application.
+     */
+    private void insertPostToDB(Post post) {
+        try {
+            
+            Document data = (new Document()
+                .append(ENTRY_ID, post.getEntryID())
+                .append(AUTHOR, post.getAuthor())
+                .append(CONTENT_BODY, post.getContent().getBody())
+                .append(ATTACHMENT_PATH, post.getContent().getAttachmentPath())
+                .append(FILE_TYPE, post.getContent().getFileType())
+                .append(POST_TITLE, post.getPostTitle())
+                .append(CATEGORY, post.getCategory())
+                .append(POSTED_DATE, post.getPostedDate())
+                .append(LAST_MODIFIED, post.getLastModifiedDate())
+                .append(LIKES, post.getLikes())
+                .append(DISLIKES, post.getDislikes())
+                .append(COMMENTS, post.getComments()) // TODO figure out type conversions if neccessary
+            );
+
+            InsertOneResult result = this.postRepository.insertOne(data);
+            System.out.println("Successfully inserted post with insert id: " + result.getInsertedId());
+        } catch (MongoException err) {
+            // TODO throws custom exceptions when they're created
+        }
+    }
+
+    /**
+     * Queries a specific post from the database.
+     * @param field - the column to to match.
+     * @param target - the target value to query for.
+     */
+    private Document queryOnePostBy(String field, String target) {
+        Bson projectionFields = Projections.fields(
+            Projections.include(field),
+            Projections.excludeId()
+        );
+        
+        Document doc = this.postRepository
+            .find(eq(field, target))
+            .projection(projectionFields)
+            .first();
+
+        return doc;
+    }
+
+    /**
+     * Queries multiple posts with given filters from the database.
+     * @param field - the column to to match.
+     * @param target - the target value to query for.
+     */
+    private MongoCursor<Document> queryMultiplePostsBy(String field, String target) {
+        Bson projectionFields = Projections.fields(
+            Projections.include(field),
+            Projections.excludeId()
+        );
+
+        // Retrieves documents that match the filter, applying a projection and a descending sort to the results
+        MongoCursor<Document> cursor = this.postRepository.find(lt(field, target))
+                .projection(projectionFields)
+                .sort(Sorts.descending(POSTED_DATE)).iterator();
+        
+        return cursor;
```

Comment:
Same question as above, do we want to return all fields in each post Document? Or just the specific field?

Reply:
We return all fields (all "attributes") of a post. For example, if we look for a post in category `general` its gonna return all posts with the `general` value in the `category` field